src/manage_nsfs/manage_nsfs_validations.js

```diff
@@ -242,19 +243,41 @@ function validate_flags_combination(type, action, input_options) {
 }
 
 /**
- * validate_flags_value_combination checks flags and value combination.
- * 1. account add or update - name should not be anonymous
+ * validate_account_name
+ * We check the name only on new accounts (account add) or accounts' rename (account update)
+ * In case, we had already an account with invalid name, it can be changed to a valid name
+ * (except anonymous which its details are only updated with its flag (and not name)
+ * (current name is not validated)
  * @param {string} type
  * @param {string} action
  * @param {object} input_options_with_data
  */
-function validate_flags_value_combination(type, action, input_options_with_data) {
+function validate_account_name(type, action, input_options_with_data) {
     if (type === TYPES.ACCOUNT) {
-        if (action === ACTIONS.ADD || action === ACTIONS.UPDATE) {
-            if (input_options_with_data.name === ANONYMOUS || input_options_with_data.new_name === ANONYMOUS) {
-                const detail = 'Account name \'anonymous\' is not valid';
+        // in name and new_name we allow type number, hence convert it to string
+        // (it is saved converted in fetch_account_data)
+        let account_name;
+        try {
+            if (action === ACTIONS.ADD) {
```

Comment:
@shirady I think that the anonymous handling is still not complete, with these changes - 
1. Account add - account name can be anonymous.
2. Account update - a regular account can be updated to be anonymous.

Reply:
Inside `validate_username` we check that the name is not anonymous
https://github.com/noobaa/noobaa-core/blob/a6e8837cb96be68fb76f8e5b007b28fe80045001/src/endpoint/iam/iam_utils.js#L420-L426
Anyway, I can leave the conditions that were there before, to be on the safe side since this name should be protected specifically.

In manual tests:
1) account add
```
➜  noobaa-core git:(nsfs-nc-system-owner) ✗ sudo node src/cmd/manage_nsfs account add --name anonymous  --new_buckets_path /tmp/nsfs_root1 --uid 1004 --gid 1004 #invalid--name
Password:
Sorry, try again.
Password:
Error accrued while getting the data from /etc/noobaa.conf.d/config_dir_redirect: Error: ENOENT: no such file or directory, open '/etc/noobaa.conf.d/config_dir_redirect'
config.get_config_root - could not find custom config_root, will use the default config_root  /etc/noobaa.conf.d
load_nsfs_nc_config.setting config.NSFS_NC_CONF_DIR /etc/noobaa.conf.d
config.load_nsfs_nc_config could not find config.json... skipping
{
  "error": {
    "code": "InvalidAccountName",
    "message": "Account name is invalid",
    "detail": "Account name 'anonymous' is not valid"
  }
}
```


2) account update
```
➜  noobaa-core git:(nsfs-iam-minor-cli-changes) sudo node src/cmd/manage_nsfs account update --name shira-1004 --new_name anonymous #invalid new_name
Error accrued while getting the data from /etc/noobaa.conf.d/config_dir_redirect: Error: ENOENT: no such file or directory, open '/etc/noobaa.conf.d/config_dir_redirect'
config.get_config_root - could not find custom config_root, will use the default config_root  /etc/noobaa.conf.d
load_nsfs_nc_config.setting config.NSFS_NC_CONF_DIR /etc/noobaa.conf.d
config.load_nsfs_nc_config could not find config.json... skipping
{
  "error": {
    "code": "InvalidAccountName",
    "message": "Account name is invalid",
    "detail": "The specified value for new_name is invalid. Should not be one of: anonymous / ."
  }
}
```

Reply:
@shirady got you, thanks for the clarification, so why we need to handle this case differently here - 
```
if (action === ACTIONS.UPDATE && input_options_with_data.name === ANONYMOUS) {
```
can't we always call `iam_utils.validate_username(account_name, 'name');` and on update of new_name check 
 `iam_utils.validate_username(account_name, 'new_name');` ?

Reply:
@romayalon I added it to be aligned with an anonymous check that also checks the `name` and not just `new_name`.
Since I saw this confusion with this change and the thoughts that I might want to handle anonymous accounts in the IAM API - I will take your advice and not touch the first function and add a new function for the names, it should be easier to understand (and maintain).

Reply:
@shirady where is the check for name as well on update? not sure I can find it

Reply:
in validate_flags_value_combination()? why not remove this function and just call iam_utils.validate_username(account_name, 'name'); as well on update?

Reply:
@romayalon we have it only in anonymous name (with the function that we had befire - untouched).
On new code we decide that we will not check it, to allow invalid account name that were already exists the ability to rename (account update with `new_name` flag).

Reply:
@shirady 
I still don't think that we want to maintain 2 functions with overlapping logic, but you will get to it later on your re-use of code of CLI and IAM PR.
I'll approve for now

Reply:
Separating them created confusion... but I can give it another try either in the PR of reuse code of IAM or in a PR with refactoring in the manage_nsfs.
I will note this as a gap in this PR description.